### PR TITLE
V3—Graph numeric scale improvements

### PR DESCRIPTION
**Pull request by bfinzer**
Added computeNiceNumericBounds in graph_utils to mimic V2 algorithm for choosing bounds from given data. (Also changed x and y to xScale and yScale in graph.tsx.)

### CODE REVIEW

File: v3/src/components/graph/hooks/graph-hooks.ts [hunk 1]
```diff
@@ -41,34 +43,42 @@ export const useGetData = (props: IUseGetDataProps) => {
...
           }
         }
+      },
+
+      setNiceDomain = (values: number[], scale: ScaleBaseType, axis: INumericAxisModel) => {
+        const valueExtent = extent(values, d => d) as [number, number],
+          niceBounds = computeNiceNumericBounds(valueExtent[0], valueExtent[1])
+        scale.domain([niceBounds.min, niceBounds.max])
+        const [min, max] = scale.domain()
+        axis.setDomain(min, max)
```
Comment (kswenson) [inline v3/src/components/graph/hooks/graph-hooks.ts:72]: I see that you're following the pattern of the preexisting code, but it should be sufficient to set the domain on the axis model and let the synchronization to the scale happen via the normal update mechanism which would simplify the code and prove that the system is working as intended.

File: v3/src/components/graph/hooks/graph-hooks.ts [hunk 2]
```diff
@@ -41,34 +43,42 @@ export const useGetData = (props: IUseGetDataProps) => {
...
+            if (xAttrId === '') {
+              xAttrId = iAttr.id
+            } else if (yAttrId === '') {
+              yAttrId = iAttr.id
+            } else {
+              break
+            }
           }
         }
+      },
```
Comment (kswenson) [inline v3/src/components/graph/hooks/graph-hooks.ts:65]: Seems to be some unnecessary indenting here.

File: v3/src/components/graph/utilities/graph_utils.ts [hunk 3]
```diff
@@ -3,8 +3,9 @@ import React from "react"
 import {defaultRadius, Rect, rTreeRect} from "../graphing-types"
 import {between} from "./math_utils"
 import {IDataSet} from "../../../data-model/data-set"
-import { ScaleBaseType } from "../models/graph-layout"
-import { prf } from "../../../utilities/profiler"
+import {ScaleBaseType} from "../models/graph-layout"
+import {prf} from "../../../utilities/profiler"
+import {isInteger} from "lodash"
```
Comment (kswenson) [inline v3/src/components/graph/utilities/graph_utils.ts:8]: Move this import to the top since it's a third-party/node_modules import.

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 Looks good -- a couple of minor suggestions. The only substantive one is to simplify `setNiceDomain()` so that it only sets the axis. The update to the D3 scale should happen automatically.

**Comment by kswenson** [comment]
Actually, one additional comment -- seems like it might be a useful to move the `setNiceDomain()` to its own utility function so it's independently testable and potentially callable from elsewhere.

File: v3/src/components/graph/hooks/graph-hooks.ts (hunk 1 above)
Comment (bfinzer) [reply v3/src/components/graph/hooks/graph-hooks.ts:72]: Yes, that's true!

File: v3/src/components/graph/hooks/graph-hooks.ts (hunk 2 above)
Comment (bfinzer) [reply v3/src/components/graph/hooks/graph-hooks.ts:65]: I don't think so. IntelliJ's formatter doesn't either.

**Pull request merged**